### PR TITLE
#{reils課題1の提出}

### PR DESCRIPTION
**Pull request by komiya7249**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,6 +27,7 @@ gem 'jbuilder', '~> 2.7'
 
 # Reduces boot times through caching; required in config/boot.rb
 gem 'bootsnap', '>= 1.4.4', require: false
+gem 'rails-i18n'
```
Comment (ibahey) [inline Gemfile:30]: rails-i18nの導入👍

File: app/assets/stylesheets/common.css
Comment (ibahey) [inline app/assets/stylesheets/common.css:1]: CSSよく書けてますね👍

File: app/controllers/schedules_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+
+  def new
+    @schedule = Schedule.new
+  
+  end
+
+  def create
+    @schedule = Schedule.new(params.require(:schedule).permit(:scheduleId,:title,:startday,:endday,:updateday,:text,:checkbox))
```
Comment (ibahey) [inline app/controllers/schedules_controller.rb]: private配下にストロングパラメータを返すメソッドを定義し、そのメソッドをよぶようにしましょう。
https://qiita.com/iverson3kobe0824/items/41b50f010c7f6b78485d

File: app/helpers/users_helper.rb
Comment (ibahey) [inline app/helpers/users_helper.rb:1]: 不要なヘルパーファイルは削除してOKです。

File: app/.DS_Store
Comment (ibahey) [inline app/.DS_Store:1]: .DS_Storeファイルは、commitに含めないようにしましょう。

File: app/models/schedule.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    validates :text,length:{maximum:500}
+    validate :endday_error
+
+    def endday_error
+        if startday.blank? || endday.blank?
+            errors.add(:endday_error,"開始日と終了日の両方を入力してください")
+        elsif startday > endday
+            errors.add(:endday_error,"終了日は開始日より後の日付を入力してください。") 
+        end
+    end
```
Comment (ibahey) [inline app/models/schedule.rb:14]: 開始日と終了日のバリデーションよく書けてますね👍

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -8,9 +8,13 @@
 
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.7.1/jquery.min.js"></script>
```
Comment (ibahey) [inline app/views/layouts/application.html.erb:11]: jQueryの導入👍

File: app/views/schedules/edit.html.erb
Comment (ibahey) [inline app/views/schedules/edit.html.erb:1]: 行末改行を入れましょう。
https://bottoms-programming.com/archives/indent-tab-space-and-new-line.html

File: app/views/schedules/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        <p><%= f.text_field:title,class:"new_field"%></p>
+        <div class="days">
+            <p class="startday"><%= f.label:startday,"開始日(必須)"%>
+            <%= f.text_field:startday%></p>
+            <p class="endday"><%= f.label:endday,"終了日(必須)"%>
+            <%= f.text_field:endday%></p>
+        </div>
+        <p>
+            <%= f.check_box:checkbox,{id: ":allday_check"},checked_value='1', unchecked_value='0'%>
+            <%= f.label:title,"終日の場合はチェックを入れます"%>
```
Comment (ibahey) [inline app/views/schedules/edit.html.erb]: f.label :allday_checkではないでしょうか。

File: app/views/schedules/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+<%= render"shared/error", obj:@schedule %>
+
+<%= form_with model: @schedule do |f|%>
+    <div class="new_form">
+        <%= f.label:title,"タイトル(必須)"%>
+        <p><%= f.text_field:title,class:"new_field"%></p>
+        <div class="days">
+            <p class="startday"><%= f.label:startday,"開始日(必須)"%>
+            <%= f.text_field:startday%></p>
```
Comment (ibahey) [inline app/views/schedules/edit.html.erb]: text_fieldですと、日付の選択がしにくいです。

File: app/views/shared/_error.html.erb
Comment (ibahey) [inline app/views/shared/_error.html.erb:1]: パーシャル化👍

File: config/locales/devise.ja.yml
Comment (ibahey) [inline config/locales/devise.ja.yml:1]: こちらのファイルも不要ですね。

File: app/views/schedules/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<h3 class="new_title">予定を編集してください</h3>
+
+<%= render"shared/error", obj:@schedule %>
+
+<%= form_with model: @schedule do |f|%>
```
Comment (ibahey) [inline app/views/schedules/edit.html.erb:5]: form_withメソッドを使用できてますね👍

File: config/routes.rb [hunk 8]
```diff
@@ -1,3 +1,8 @@
 Rails.application.routes.draw do
+  get 'schedules/index',to:'schedules#index'
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+
+  resources :schedules
```
Comment (ibahey) [inline config/routes.rb:5]: resourcesメソッドを使用できてますね👍

File: app/assets/stylesheets/common.css [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+  background-color: paleturquoise;
+  width: 30%;
+  font-size: 17px;
+  line-height: 35px;
+  border: 1px solid teal;
+}
+
+.show_td {
+  border: 1px solid teal;
+}
```
Comment (TakuyaChida) [inline app/assets/stylesheets/common.css:263]: good-badge
共通のCSSがまとめられていていいですね！

File: app/controllers/schedules_controller.rb [hunk 10]
```diff
@@ -0,0 +1,60 @@
+class SchedulesController < ApplicationController
+  def index
+    @schedules = Schedule.all
+  end
+  
+
```
Comment (TakuyaChida) [inline app/controllers/schedules_controller.rb:6]: nits-badge
next-badge
改行が1行だったり、2行だったり、3行だったりと統一性がみられないので、
1行で統一するようにしましょう！
次回から気をつけてください！

File: app/controllers/schedules_controller.rb [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    if @schedule.save
+      flash[:notice] = "予定を登録しました"
+      redirect_to :schedules
+    else
+      flash[:notice_error] = "予定の登録に失敗しました"
+      render"new"
+    end
+    
+
+
```
Comment (TakuyaChida) [inline app/controllers/schedules_controller.rb:23]: nits-badge
next-badge
改行が1行だったり、2行だったり、3行だったりと統一性がみられないので、
1行で統一するようにしましょう！
次回から気をつけてください！

**Review by TakuyaChida — APPROVED** [review]
確認させていただきました！

丁寧に作成できていていいですね！
前回の指摘事項の修正と、課題の要件が満たされていることを確認しましたのでLGTMです！

別途気になった点ですが、
・PR作成時には、Descriptionの項目を記載できるようにしましょう！
現在は空欄となってしまっているので、作成したアプリケーションの概要や、特にレビューしてほしい点などを記載できると尚良いです！

・コミット時のコメントや粒度を改善しましょう！
もう少し細かくコミットできると尚良いです！
コミット時のコメントもできるだけ具体的にするように意識しましょう。
[コミットについて](https://qiita.com/jnchito/items/40e0c7d32fde352607be)
